### PR TITLE
Extract ad campaign from Play Store referrer

### PR DESCRIPTION
**Pull request by CDRussell**
Task/Issue URL: https://app.asana.com/0/488551667048375/1207902032895525/f 

### Description
Add ability to extract an ad campaign from Play Store referrer URLs.

### Steps to test this PR

ℹ️ Logcat filter: `Pixel sent: m_android_install`
ℹ️ This requires the Play Store to be available and active on your test device / emulator.
⚠️  You need to remove the **production** app from your device during this test

**Local changes for testing**
it's a nightmare to E2E test this because it involves first having released these changes in an app version on the Play Store, so that the app you install from the Play Store referral link has the changes already in it, but we can't release until we've tested the changes 🐔/🥚 . there is a neat hack to simulate this workflow though

- [x] delete the line `applicationIdSuffix ".debug"` from your `build.gradle`
- [x] Uninstall the prod app if already installed

**Send a Play Store app install link containing referrer to your test device / emulator**
- [x] Send a URL to your device so that you can click on it (e.g., via email). The format should be like this:
  - ```https://play.google.com/store/apps/details?id=com.duckduckgo.mobile.android&referrer=origin%3Dfunnel_playstore_whatever```
  - You can change the value to test different things, including adding extra parts afterwards but it must start `origin%3D`
- [x] Click on the link to launch the app's listing in the Play Store app. **Don't install it**

**Install the APK from this branch (Play variant)**
- [x] Now you can install the APK from this branch `./gradlew installPlayDebug`
- [x] Launch the app
- [x] Verify in logcat that: install pixel contains ad campaign you specified in the URL above
- [x] Verify pixel params look right, including `reinstall` value
- [x] Close the app and relaunch; verify not sent again

**Testing what happens if referrer does not contain origin (when not installed from Play Store)**
- [x] Delete the app
- [x] Install from this branch `./gradlew installPlayDebug` (without visiting the referral link) and launch the app
- [x] Verify in logcat that: install pixel still fires but that there is no origin attached

**Testing what happens if referrer does not contain origin (when installed from Play Store)**
- [x] Requires local hack: hardcode `VerificationCheckPlayStoreInstallImpl.installedFromPlayStore()` to return `true`
- [x] Delete the app
- [x] Install from this branch `./gradlew installPlayDebug` (without visiting the referral link) and launch the app
- [x] Verify in logcat that: install pixel still fires and that origin has been set to default, `origin=funnel_playstore`

**Testing reinstall flag picked up correctly**
- [x] Requires local hack: hardcode `ReinstallAtbListener.beforeAtbInit()` to detect returning user
- [x] Delete the app
- [x] `./gradlew installPlayDebug` (don't need to visit the referral link) and launch the app
- [x] Verify in logcat that: install pixel fires and that `reinstall=true`
- [x] Repeat the above but hardcoding returning user to false, and verify `reinstall=false`

**Testing non-Play variant**
- [x] Delete app
- [x] `./gradlew installInternalDebug` and launch app
- [x] Verify in logcat that: install pixel **not** fired

### CODE REVIEW

**Comment by CDRussell** [comment]
* **#4818** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4818?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CDRussell and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: common/common-utils/src/main/java/com/duckduckgo/common/utils/playstore/PlayStoreUtils.kt [hunk 1]
```diff
@@ -42,7 +42,7 @@ class PlayStoreAndroidUtils @Inject constructor(val context: Context) : PlayStor
             val installSource = context.packageManager.getInstallerPackageName(DDG_APP_PACKAGE)
             return matchesPlayStoreInstallSource(installSource)
         } catch (e: IllegalArgumentException) {
-            Timber.w(e, "Can't determine if app was installed from Play Store; assuming it wasn't")
+            Timber.w("Can't determine if app was installed from Play Store; assuming it wasn't")
```
Comment (CDRussell) [inline common/common-utils/src/main/java/com/duckduckgo/common/utils/playstore/PlayStoreUtils.kt:45]: This is a noisy log we only see during development; removing the stacktrace

**Review by malmstein — APPROVED** [review]
Works as expected, thanks @CDRussell !

**Comment by CDRussell** [comment]
### Merge activity

* **Aug 5, 11:07 AM EDT**: @CDRussell started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/4818).

**Pull request merged**